ote_sdk/ote_sdk/entities/annotation.py

```diff
@@ -26,7 +31,15 @@ class Annotation(metaclass=abc.ABCMeta):
     def __init__(
         self, shape: ShapeEntity, labels: List[ScoredLabel], id: Optional[ID] = None
     ):
+
         self.__id_ = ID(ObjectId()) if id is None else id
+        check_input_param_type(
+            RequiredParamTypeCheck(shape, "shape", ShapeEntity),
```

Comment:
I need second str parameter for ValueError exception message, to give information which parameter has unexpected type

Reply:
Can we avoid this duplication somehow? Get str by variable or get variable by str?

Reply:
thanks, I will try to do it

Reply:
If I try to get parameter_name inside the ote_sdk/ote_sdk/utils/argument_checks.py classes, it will be already changed to the “parameter”, so I have to pass str values directly in the ote_sdk classes and functions. Ex., if I change RequiredParamTypeCheck __init__ method to:
self.parameter = parameter
self.parameter_name = f"{parameter=}".split("=")[0]
self.expected_type = expected_type
Then parameter_name attribute will be equal to the “parameter”
Is there better way to get parameter name?

Reply:
Thanks, added InputParamTypeCheck class, removed RequiredParamTypeCheck and OptionalParamTypeCheck classes